lib/index.js

```diff
@@ -38,11 +38,27 @@ const create = function (servicename, overrides = {}) {
     config.agent = Agent.create(config.protocol, config)
   }
 
-  const client = new Client(servicename, config)
+  // If overrides are passed do not cache the client
+  if (Object.keys(overrides).length) {
+    const client = new Client(servicename, config)
 
-  debug('create(): created client id %s for service %s', client.id, servicename)
+    debug('create(): created client id %s for service %s', client.id, servicename)
 
-  return client
+    return client
+  }
+
+  // If the client is not already cached create an instance and cache
+  if (!Object.keys(overrides).length && typeof GlobalConfig.clientInstances[servicename] === 'undefined') {
```

Comment:
`!Object.keys(overrides).length` is redundant now :)

Reply:
🤦‍♂ I rewrote that section several times trying to simplify it too.